lib/bloom_remit_client/requests/url_builder.rb

```diff
@@ -0,0 +1,56 @@
+module BloomRemitClient
+  module Requests
+    class UrlBuilder
+      require 'addressable/uri'
+      include Virtus.model
+      include ActiveModel::Validations
+
+      attribute :scheme, String, default: 'https'
+      attribute :subdomain, String
+      attribute :userinfo, String
+      attribute :path, String
+      attribute :host, String
+      attribute :port, Integer
+      attribute :query, String
+      attribute :query_params, Hash, default: {}
+      attribute :fragment, String
+
+      validates :path, :host, :scheme, presence: true
+      validates :scheme, inclusion: ['https', 'http']
+
+      def call
+        builder.build(to_h).to_s
+      end
+
+      def host
+        return super unless subdomain.present?
+        "#{subdomain}.#{super}"
+      end
+
+      def path
```

Comment:
@ramontayag what do u mean? Basically, default value for path should be empty line... This override is just normalizer which adds `/` to path string beginning if it not present(`path/without/slash` => `/path/without/slash`) But looks like better to override setter:
```ruby
def path=(value)
  @path = normalize_path(value) 
end
```